### ISSUE TITLE
Filter out sequences matching specific patterns at demultiplexing

### ISSUE THREAD

**Issue by cnluzon** [elsasserlab/minute#32]
Besides the adapter sequences mentioned in #21, there are cases where we want to filter out sequences matching a specific pattern.

Right now we have a couple candidate sequences we would like to take care of, 1) adapters and 2) AT-rich sequence: `TTTTTCTTTTCTTTTTTCTTTTCCTTCCTTCTAA`.

Adapters are not so problematic because they do not map (although ideally we would remove both mates in this case, whereas mapping we may get one of the mates mapped). AT-rich sequence does map to repetitive regions generating an artifactual peak that is usually taken care of on normalization (if there is input, it's present just the same in the input sample). On average profiles, it still generates noise when looking at repetitive regions, though.

We have seen that filtering it out on demultiplexing solves this problem so far. It seems to be a method-specific artifact that we want to monitor in every run, so we can also check if this filtering has some side effect. What we would like would be filtering out these and counting how many reads were discarded. Perhaps even keeping those reads in a file for further inspection.

So, basically, the approach would be:

1. Specify some kind of "sequence blacklist" in the config to filter out at sequence level.
2. Filter for these when demultiplexing.
3. Generate stats on this.

Some uncertainty is to be considered in the matching (current "garage" solution allows for 1 mismatch). Is this something `cutadapt` does? 

A very sequence-sensitive way of doing this would be mapping the adaptor sequence to each reads and have a score threshold for this, but I'm not sure if that is the most efficient/adequate approach.

Otherwise we could just implement this ad-hoc 1-mismatch approach, which is very stringent with the amount of reads it will remove and seems to produce the desired effect.
